python/src/iceberg/schema.py

```diff
@@ -67,6 +67,17 @@ def __repr__(self):
             f"Schema(fields={repr(self.columns)}, schema_id={self.schema_id}, identifier_field_ids={self.identifier_field_ids})"
         )
 
+    def __eq__(self, other) -> bool:
```

Comment:
I think we also need to check the identifier fields. Java's `Schema::isSameSchema` does so.

https://github.com/apache/iceberg/blob/06cdfa3f7c4ebd26766bc67100ba5d30b6bf3940/api/src/main/java/org/apache/iceberg/Schema.java#L444-L452

Reply:
This is now also implemented using the autogenerated `__eq__` of the dataclass 👍🏻

Reply:
Does this `__eq__` implementation override the autogenerated one? If we're using a generated one, I don't understand why this would still be needed.

Reply:
Yes, that's true. I forgot to remove this one, thanks!

Reply:
I was a bit too quick on this one. Schema doesn't have an autogenerated one, but the types do have one now. So we still need the eq on the schema

Reply:
In that case, I think we'll need to check the identifier field IDs like Kyle suggested

Reply:
Ah, right! Sorry about that